mono/mini/method-to-ir.c

```diff
@@ -9660,13 +9660,13 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 					ins->type = STACK_OBJ;
 					ins->klass = klass;
 					ins->inst_c0 = val->inst_c0;
-					ins->dreg = alloc_dreg (cfg, val->type);
+					ins->dreg = alloc_dreg (cfg, (MonoStackType)val->type);
```

Comment:
In time we might be able to change type to be MonoStackType, but this is probably the better fix for now.

As well, consider for now changing alloc_dreg to take int -- there is no type safety gained in C either way, and we aren't aiming for any gains from C++ yet, just correctness. Making it "worse C++" and "equally good C" is ok for now.